cpp/models/abm/world.cpp

```diff
@@ -38,20 +34,37 @@ namespace abm
 LocationId World::add_location(LocationType type, uint32_t num_cells)
 {
     LocationId id = {static_cast<uint32_t>(m_locations.size()), type};
-    m_locations.emplace_back(std::make_unique<Location>(id, parameters.get_num_groups(), num_cells));
+    m_locations.emplace_back(id, parameters.get_num_groups(), num_cells);
     m_has_locations[size_t(type)] = true;
+
+    // mark caches for rebuild
+    m_is_local_population_cache_valid = false;
+    m_are_exposure_caches_valid       = false;
+    m_exposure_caches_need_rebuild    = true;
+
     return id;
 }
 
-Person& World::add_person(const LocationId id, AgeGroup age)
+PersonId World::add_person(const LocationId id, AgeGroup age)
+{
+    return add_person(Person(m_rng, id, age));
+}
+
+PersonId World::add_person(Person&& person)
 {
-    assert(age.get() < parameters.get_num_groups());
-    uint32_t person_id = static_cast<uint32_t>(m_persons.size());
-    m_persons.push_back(std::make_unique<Person>(m_rng, get_individualized_location(id), age, person_id));
-    auto& person = *m_persons.back();
-    person.set_assigned_location(m_cemetery_id);
-    get_individualized_location(id).add_person(person);
-    return person;
+    assert(person.get_location().index != INVALID_LOCATION_INDEX);
+    assert(person.get_location().index < m_locations.size());
+    assert(person.get_age().get() < parameters.get_num_groups());
```

Comment:
I think messages like that can be useful, but mostly on more complicated asserts, or when the message content could help avoid using a debugger. Since these are pretty simple (bound) checks, I don't think the messages are needed.

If you are undecided, then I would not add them. But I will, if you think they are beneficial.